File: php/source/dexi/Dexi.php
<?php

require_once 'controllers/ExecutionsController.php';
require_once 'controllers/RunsController.php';

class Dexi {
    private $accountId;
    private $apiKey;

    /**
     * Create new API client instance
     * @param $accountId
     * @param $apiKey
     */
    public function __construct($accountId, $apiKey) {
        $this->accountId = $accountId;
        $this->apiKey = $apiKey;
    }


    /**
     * Interact with executions using the API
     *
     * @return ExecutionsController
     */
    public function executions() {
        return new ExecutionsController($this->accountId, $this->apiKey);
    }

    /**
     * Interact with runs using the API
     *
     * @return RunsController
     */
    public function runs() {
        return new RunsController($this->accountId, $this->apiKey);
    }
}

File: dotnet/source/dexi/Dexi.cs
namespace Dexi {

    public class Dexi {
        private String accountId;
        private String apiKey;

        public Dexi(String accountId, String apiKey) {
            this.accountId = accountId;
            this.apiKey = apiKey;
        }

        public ExecutionsController executions() {
            return new ExecutionsController(accountId, apiKey);
        }

        public RunsController runs() {
            return new RunsController(accountId, apiKey);
        }

    }
}